lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -0,0 +1,124 @@
+        #             Multiple capital letters together
+        catName = str(schemaTypeName[0])
+        nbChars = len(schemaTypeName)
+        for i in range(1, nbChars):
+            if schemaTypeName[i].isupper() and not schemaTypeName[i-1].isdigit():
+                if (i < (nbChars-1)) and not schemaTypeName[i+1].isupper():
+                    catName = catName + str(' ') + str(schemaTypeName[i])
+                else:
+                    catName = catName + str(schemaTypeName[i])
+            elif schemaTypeName[i] == '_':
+                continue
+            else:
+                catName = catName + str(schemaTypeName[i])
+
+        # Make each word start with an uppercase.
+        catName = catName.title()
+
+        return catName
+
+    def createTransformAttributesSection(self, ufeSceneItem, sectionName, attrsToAdd):
+        # Get the xformOp order and add those attributes (in order)
+        # followed by the xformOp order attribute.
+        allAttrs = self.attrS.attributeNames
+        prim = mu.getPrimFromRawItem(ufeSceneItem.getRawAddress())
+        geomX = UsdGeom.Xformable(prim)
+        xformOps = geomX.GetOrderedXformOps()
+        xformOpOrderNames = [op.GetOpName() for op in xformOps]
+        xformOpOrderNames.append(UsdGeom.Tokens.xformOpOrder)
+        # Don't use createSection because we want a sub-sections.
+        with ufeAeTemplate.Layout(self, sectionName):
+            with ufeAeTemplate.Layout(self, 'Transform Attributes'):
+                attrsToAdd.remove(UsdGeom.Tokens.xformOpOrder)
+                self.addControls(xformOpOrderNames)
+
+                # Get the remainder of the xformOps and add them in an Unused section.
+                xformOpUnusedNames = fnmatch.filter(allAttrs, 'xformOp:*')
+                xformOpUnusedNames = [ele for ele in xformOpUnusedNames if ele not in xformOpOrderNames]
+                self.createSection('Unused Transform Attributes', xformOpUnusedNames, collapse=True)
+
+            # Then add any reamining Xformable attributes
+            self.addControls(attrsToAdd)
+
+    def buildUI(self, ufeSceneItem):
+        usdSch = Usd.SchemaRegistry()
+
+        # We use UFE for the ancestor node types since it caches the
+        # results by node type.
+        for schemaType in ufeSceneItem.ancestorNodeTypes():
+            schemaType = usdSch.GetTypeFromName(schemaType)
+            schemaTypeName = schemaType.typeName
+            sectionName = self.sectionNameFromSchema(schemaTypeName)
+            if schemaType.pythonClass:
+                attrsToAdd = schemaType.pythonClass.GetSchemaAttributeNames(False)
+
+                # We have a special case when building the Xformable section.
+                if schemaTypeName == 'UsdGeomXformable':
+                    self.createTransformAttributesSection(ufeSceneItem, sectionName, attrsToAdd)
+                else:
+                    self.createSection(sectionName, attrsToAdd)
```

Comment:
This method completely changed from ae/base/ae_template.py. We no longer have hard-coded sections and attributes. Instead using the ancestor node types I get all the schemas and for each one get the attributes.

The only exception is the Xformable, where we still display the transform attributes by xformoporder.